CHANGELOG.md

```diff
@@ -3,7 +3,10 @@
 BUG FIXES:
 
  * core: Fixed a bug that only ran a task `shutdown_delay` if the task had a registered service [[GH-7663](https://github.com/hashicorp/nomad/issues/7663)]
+ * core: Fixed a panic when deleting a job that had previously been updated. [[GH-7758](https://github.com/hashicorp/nomad/issues/7758)]
```

Comment:
I'd use garbage collection or purging here - as we don't talk about jobs deleting much.  Maybe something like this.  Also, we sort by line length within the same component, so this can go first in the bug fixes section.

```suggestion
 * core: Fixed a panic when garbage collecting a job with allocations spanning multiple versions. [[GH-7758](https://github.com/hashicorp/nomad/issues/7758)]
```

Reply:
Done.